plugins/lang/python/python_loader.cc

```diff
@@ -634,16 +637,62 @@ class mldb_wrapper(object):
             if res.wasSuccessful():
                 self.script.set_return("success")
 
+
 class MldbUnitTest(unittest.TestCase):
-    def _assert_flat_result(self, res, expected):
-        if res[0][0] != '_rowName':
-            raise Exception("Cannot order rows if header is missing from "
-                            "response")
-        if expected[0][0] != '_rowName':
-            raise Exception("The first row of expected needs to be the header")
+    import json
+
+    longMessage = True # Appends the user message to the normal message
+
+    class _AssertMldbRaisesContext(object):
+        """A context manager used to implement TestCase.assertRaises* methods.
+        Inspired from python unittests.
+        """
+
+        def __init__(self, test_case, expected_regexp=None, status_code=None):
+            self.failureException = test_case.failureException
+            self.expected_regexp = expected_regexp
+            self.status_code = status_code
+
+        def __enter__(self):
+            return self
+
+        def __exit__(self, exc_type, exc_value, tb):
+            if exc_type is None:
+                raise self.failureException(
+                    "{0} not raised".format(mldb_wrapper.MldbBaseException))
+            if not issubclass(exc_type, mldb_wrapper.MldbBaseException):
+                # let unexpected exceptions pass through
+                return False
+            self.exception = exc_value # store for later retrieval
+            if self.expected_regexp:
+                import re
+                expected_regexp = self.expected_regexp
+                if isinstance(expected_regexp, basestring):
+                    expected_regexp = re.compile(expected_regexp)
+                if not expected_regexp.search(str(exc_value.response.text)):
+                    raise self.failureException('"%s" does not match "%s"' %
+                        (expected_regexp.pattern,
+                         str(exc_value.response.text)))
+            if self.status_code:
+                if exc_value.response.status_code != self.status_code:
+                    raise self.failureException(
+                        "Status codes are not equal: {} != {}".format(
+                            exc_value.response.status_code, self.status_code))
+            return True
+
+    def _get_base_msg(self, res, expected):
+        return '{line}Result: {res}{line}Expected: {expected}'.format(
+            line='\n' + '*' * 10 + '\n',
+            res=MldbUnitTest.json.dumps(res, indent=4),
+            expected=MldbUnitTest.json.dumps(expected, indent=4))
+
+    def assertTableResultEquals(self, res, expected, msg=""):
```

Comment:
basically yes. we have a special helper for format=table because that's the default output of mldb.query and we have a special helper for format=full because that's the most verbose one. Everything else can still be tested "the old way" but we can always add more helpers.